src/internal/analytics-metadata/__tests__/labels-utils.test.tsx

```diff
@@ -0,0 +1,228 @@
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('returned label');
+  });
+  test('returns aria label of aria-labelledby element', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55:">
+          content
+        </div>
+        <div id="id:r55:" aria-label="labelled by label">
+          content
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('labelled by label');
+  });
+  test('returns text content of aria-labelledby element', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55:">
+          content
+        </div>
+        <div id="id:r55:">second content</div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('second content');
+  });
+  test('returns text content of first aria-labelledby element', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55: id:r56:">
+          content
+        </div>
+        <div id="id:r55:">second content</div>
+        <div id="id:r56:">third content</div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('second content');
+  });
+  test('returns empty string if aria-labelledby element does not exist', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55: id:r56:">
+          content
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('');
+  });
+});
+
+describe('processLabel', () => {
+  test('returns an empty string when element and/or identifier are null', () => {
+    const { container } = render(
+      <div>
+        <div id="target">content</div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(null, null)).toEqual('');
+    expect(processLabel(target, null)).toEqual('');
+    expect(processLabel(null, 'selector')).toEqual('');
+  });
+  test('returns an empty string when the selector cannot be resolved to an element', () => {
+    const { container } = render(
+      <div>
+        <div id="target">content</div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(target, 'invalid-selector')).toEqual('');
+  });
+  test('returns node label when selector is an empty string', () => {
+    const { container } = render(
+      <div>
+        <div id="target">content</div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(target, '')).toEqual('content');
```

Comment:
Why do we get the content back here but if you provide a null selector as above then you get back an empty string?

Reply:
because an empty string selector is a shortcut for the current element